sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -442,17 +442,22 @@ object DateTimeUtils {
 
   /**
    * Trims and parses a given UTF8 string to a corresponding [[Long]] value which representing the
-   * number of microseconds since the epoch. The result is independent of time zones,
-   * which means that zone ID in the input string will be ignored.
+   * number of microseconds since the epoch. The result will be independent of time zones.
+   *
+   * If the input string contains a component associated with time zone, the method will return
+   * `None` if `ignoreTimeZone` is set to `false`. If `ignoreTimeZone` is set to `true`, the method
+   * will simply discard the time zone component. Enable the check to detect situations like parsing
+   * a timestamp with time zone as TimestampNTZType.
+   *
    * The return type is [[Option]] in order to distinguish between 0L and null. Please
    * refer to `parseTimestampString` for the allowed formats.
    */
-  def stringToTimestampWithoutTimeZone(s: UTF8String): Option[Long] = {
+  def stringToTimestampWithoutTimeZone(s: UTF8String, ignoreTimeZone: Boolean): Option[Long] = {
```

Comment:
The parameter `ignoreTimeZone` is confusing. 
Let's use `failOnError` to keep consistency with other ANSI related code.

Reply:
I understand, but we don't fail if ignoreTimeZone was set to `false`, we return an Option instead. I can update in TimestampFormatter but `failOnError` in `stringToTimestampWithoutTimeZone` might be a bit misleading. Let me know what you think. Thanks.

Reply:
Updated.